### ISSUE TITLE
React-icons DLL error

### ISSUE THREAD

**Issue by xtrinch** [mxstbr/react-boilerplate#1405]
I'm having trouble with react-icons and webpack DLL:

```
$ npm run build:dll

> react-boilerplate@3.3.3 build:dll /home/trinch/**
> node ./internals/scripts/dependencies.js

Building the Webpack DLL...
Hash: 7b4274842c485157f2ba
Version: webpack 2.1.0-beta.25
Time: 4410ms
                      Asset     Size  Chunks             Chunk Names
reactBoilerplateDeps.dll.js  3.41 MB       0  [emitted]  reactBoilerplateDeps
chunk    {0} reactBoilerplateDeps.dll.js (reactBoilerplateDeps) 2.74 MB [entry] [rendered]
    [2] dll reactBoilerplateDeps 12 bytes {0} [built]
     + 808 hidden modules

ERROR in dll reactBoilerplateDeps
Module not found: Error: Can't resolve 'react-icons' in '/home/trinch/**'
 @ dll reactBoilerplateDeps
Done in 34.43s.

```

I installed react-icons via yarn add.
I don't see any errors on my webpage and I can use the icons without a problem.
What is that error up there, then?

**Comment by manolkalinov** [comment]
Hey @xtrinch, go to internals/config.js and find exclude:, under sanitize.css add react-icons.

**Comment by xtrinch** [comment]
@manolkalinov Cool, that works, thanks! Quick explanation why do I need to exclude it, in what way is it not intended for the browser?

**Comment by activedecay** [comment]
I don't understand this error below. I have followed the above directions. Thanks for any help :) 
```
[nodemon] starting `node server`
start! ✓

webpack built 9a2a6b755a3f12ffd33e in 4880ms

ERROR in ./~/react-icons/fa/beer.js
Module parse failed: /home/jfaust/hack/musical-meme/node_modules/react-icons/fa/beer.js Unexpected token (6:4)
You may need an appropriate loader to handle this file type.
| 
| const FaBeer = props => (
|     <Icon viewBox="0 0 40 40" {...props}>
|         <g><path d="m14.8 20v-8.6h-5.7v5.7q0 1.2 0.8 2.1t2 0.8h2.9z m22.8 10v4.3h-25.7v-4.3l2.9-4.3h-2.9q-3.5 0-6-2.5t-2.5-6.1v-7.1l-1.5-1.4 0.7-2.9h10.8l0.7-2.8h21.4l0.7 4.2-1.4 0.8v17.8z"/></g>
|     </Icon>
 @ ./app/containers/Welcome/index.js 49:0-41
 @ ./app/routes.js
 @ ./app/app.js
 @ multi main
```

**Comment by activedecay** [comment]
found the fix! it's working.

> If you are not using es6 compiler like babel or rollup.js, it's possible to include icons from the compiled folder ./lib. Babel by default will ignore node_modules so if you don't want to change the settings you will need to use files from ./lib
> 
> var FaBeer = require('react-icons/lib/fa/beer');
 
RTFM: http://gorangajic.github.io/react-icons/

**Comment by coleclayman** [comment]
I ended up trying @manolkalinov  's solution of adding a list of `exclude`, but it still didn't work – I found another suggestion on a different GH issue and this is what ended up working for me.

In your `webpack.config.*.js` file, add the following to `plugins` after `new webpack.DllPlugin({})`:

`new webpack.IgnorePlugin(/your_package_name_here/)`. For me, it was yarn that was blowing stuff up.

Hope this helps!
